tests/test_subdirectory.py

```diff
@@ -0,0 +1,176 @@
+import os
+
+from plumbum import local
+from plumbum.cmd import git
+
+import copier
+from copier.config import make_config
+from copier.main import update_diff
+
+
+def git_init(message="hello world"):
+    git("init")
+    git("config", "user.name", "Copier Test")
+    git("config", "user.email", "test@copier")
+    git("add", ".")
+    git("commit", "-m", message)
+
+
+def test_copy_subdirectory_api_option(tmp_path):
+    copier.copy(
+        "./tests/demo_subdirectory", tmp_path, force=True, subdirectory="api_project"
+    )
+    assert (tmp_path / "api_readme.md").exists()
+
+
+def test_copy_subdirectory_config(tmp_path):
+    copier.copy("./tests/demo_subdirectory", tmp_path, force=True)
+    assert (tmp_path / "conf_readme.md").exists()
+
+
+def test_update_subdirectory(tmp_path):
+    copier.copy("./tests/demo_subdirectory", tmp_path, force=True)
+
+    with local.cwd(tmp_path):
+        git_init()
+
+    conf = make_config("./tests/demo_subdirectory", str(tmp_path), force=True)
+    update_diff(conf)
+    assert not (tmp_path / "conf_project").exists()
+    assert not (tmp_path / "api_project").exists()
+    assert not (tmp_path / "api_readme.md").exists()
+    assert (tmp_path / "conf_readme.md").exists()
+
+
+def test_new_version_uses_subdirectory(tmp_path_factory):
+    # Template in v1 doesn't have a _subdirectory;
+    # in v2 it moves all things into a subdir and adds that key to copier.yml.
+    # Some files change. Downstream project has evolved too. Does that work as expected?
+    template_path = tmp_path_factory.mktemp("subdirectory_template")
+    project_path = tmp_path_factory.mktemp("subdirectory_project")
+
+    # First, create the template with an initial README
+    with local.cwd(template_path):
+        with open("README.md", "w") as fd:
+            fd.write("upstream version 1\n")
+
+        with open("[[_copier_conf.answers_file]].tmpl", "w") as fd:
+            fd.write("[[_copier_answers|to_nice_yaml]]\n")
+
+        git_init("hello template")
+
+    # Generate the project a first time, assert the README exists
+    copier.copy(str(template_path), project_path, force=True)
+    assert (project_path / "README.md").exists()
```

Comment:
Then check those are used:

```suggestion
    assert (project_path / "README.md").exists()
    assert "_commit: v1" in (project_path / ".copier-answers.yml").read_text()
```